cmd/ping/main.go

```diff
@@ -15,25 +15,31 @@
 package main
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"log"
 	"net/http"
 	"os"
 	"sync"
+	"time"
 
 	"github.com/GoogleCloudPlatform/gcping/internal/config"
+	"github.com/patrickmn/go-cache"
 )
 
 var once sync.Once
 
 func main() {
+	ctx := context.Background()
```

Comment:
Because of the way you structured the program you will probably not need this, but for several long running operations that all take `context.Context` to synchronize work this is the common pattern

```
ctx, cancel := context.WithCancel(context.Background())
defer cancel()
signalCh := make(chan os.Signal, 1)
signal.Notify(signalCh, os.Interrupt)
go func() {
	sig := <-signalCh
	fmt.Printf("termination signal received: %s", sig)
	cancel()
}()
```

followed by 

```
go func1(ctx)  // Concurrent
go func2(ctx)  // Concurrent
func3(ctx) //       <-- this one blocks
```

That way when `Ctrl + C` is sent then all the contexts are cancelled and gracefully cleaned up

This can also be used with the `errgroup` package (golang.org/x/sync/errgroup).

So that second block of code would turn into:
```
group, ctx := errgroup.WithContext(ctx)

group.Go(func1(ctx))

group.Go(func() error {
	return func2(ctx)
})


group.Go(func() error {
	return func3(ctx)
})
err := group.Wait()
log.Fatal(err)
```

Like I said this is not needed for this code, but future Go code may need to leverage this pattern.